### DISCUSSION TITLE
ER_BAD_FIELD_ERROR: Unknown column 'ts' in 'field list'

### DISCUSSION THREAD

**Discussion by LeadcodeDev** [adonisjs/core#1902] [Help] [answered]
Hello!

I'm trying to create a resource within my API. During registration, I get this error 500 in the API console (adonis5).

I don't have any 'ts' column in my application, I don't see where this error can come from...

If you want to look at my files, tell me which ones you want...

Thanks in advance

![image](https://user-images.githubusercontent.com/8946317/98972803-3a70a800-2513-11eb-8968-0eb2e52aa6b1.png)

**Comment by LeadcodeDev** [comment]
My migration :
```ts
public async up() {
		this.schema.createTable(this.tableName, (table) => {
			table.increments('id')
			table.string('label')
			table.boolean('active').defaultTo(false)
			table.dateTime('start').notNullable()
			table.dateTime('expiration').notNullable()
			table.boolean('has_max_uses').defaultTo(false)
			table.integer('uses').defaultTo(0)
			table.boolean('has_new_reduce').defaultTo(false)
			table.float('new_reduce').defaultTo(0)
			table.string('new_reduce_unit').notNullable()
			table.float('reduce').defaultTo(0)
			table.string('reduce_unit').notNullable()
			table.timestamps(true, true)
		})
	}
```

My model :
```ts
export default class Promotion extends BaseModel {
	@column({ isPrimary: true })
	public id: number

	@column()
	public label: string

	@column()
	public active: boolean

	@column()
	public start: DateTime

	@column()
	public expiration: DateTime

	@column()
	public hasMaxUses: boolean

	@column()
	public uses: number

	@column()
	public hasNewReduce: boolean

	@column()
	public newReduce: number

	@column()
	public newReduceUnit: string

	@column()
	public reduce: number

	@column()
	public reduceUnit: string

	@column.dateTime({ autoCreate: true })
	public createdAt: DateTime

	@column.dateTime({ autoCreate: true, autoUpdate: true })
	public updatedAt: DateTime
}
```

**Reply by hariprasath-yadav** [reply]
Share the exact statement you used to insert the record

**Reply by LeadcodeDev** [reply]
My console.log() on my nuxt
```
data {
  label: 'Label',
  active: true,  
  start: DateTime {
    ts: 1605168720000,
    _zone: LocalZone {},
    loc: Locale {
      locale: 'fr-FR',
      numberingSystem: null,
      outputCalendar: null,
      intl: 'fr-FR',
      weekdaysCache: [Object],
      monthsCache: [Object],
      meridiemCache: null,
      eraCache: {},
      specifiedLocale: null,
      fastNumbersCached: null
    },
    invalid: null,
    weekData: null,
    c: {
      year: 2020,
      month: 11,
      day: 12,
      hour: 9,
      minute: 12,
      second: 0,
      millisecond: 0
    },
    o: 60,
    isLuxonDateTime: true
  },
  expiration: DateTime {
    ts: 1605341760000,
    _zone: LocalZone {},
    loc: Locale {
      locale: 'fr-FR',
      numberingSystem: null,
      outputCalendar: null,
      intl: 'fr-FR',
      weekdaysCache: [Object],
      monthsCache: [Object],
      meridiemCache: null,
      eraCache: {},
      specifiedLocale: null,
      fastNumbersCached: null
    c: {
    c: {
      year: 2020,
      month: 11,
      day: 14,
      hour: 9,
      minute: 16,
      second: 0,
      millisecond: 0
    },
    o: 60,
    isLuxonDateTime: true
  },
  hasMaxUses: true,
  uses: 10,
  hasNewReduce: true,
  newReduce: 5,
  newReduceUnit: '%',
  reduce: 20,
  reduceUnit: '%'
}
```

The error :
```
[1605255192053] ERROR    (api/17020 on DESKTOP-9Q3Q14G): insert into `promotions` (`active`, `created_at`, `expiration`, `has_max_uses`, `has_new_reduce`, `label`, `new_reduce`, `new_reduce_unit`, `reduce`, `reduce_unit`, `start`, `updated_at`, `uses`) values (true, '2020-11-13 09:13:12', 2020-11-14 09:16:00.000 +01:00, true, true, 'Label', 5, '%', 20, '%', 2020-11-12 09:12:00.000 +01:00, '2020-11-13 09:13:12', 10) - ER_BAD_FIELD_ERROR: Unknown column 'ts' in 'field list'
```

**Comment by hariprasath-yadav** [comment]
what is your lucid version

**Reply by LeadcodeDev** [reply]
I have a lasted version :
```json
"@adonisjs/lucid": "^9.0.3"
```

**Reply by hariprasath-yadav** [reply]
previous i used this package only but it gives some error so i moved to

> @adonisjs/lucid@^8.3.0

i think it will help because version 9.0.3 shows yours kind of errors to me like date function error
after downgrade, it works fine so try this by

> npm i --save @adonisjs/lucid@^8.3.0

**Reply by LeadcodeDev** [reply]
I can't downgrade without downgrade all dependencies :'(

**Reply by edesideri** [reply]
i solved same issue using Date object instead luxon DateTime.
example:
`await Database
                    .table('password_resets')
                    .insert({ email: user.email, token: newToken, created_at: new Date() })`

**Reply by LeadcodeDev** [reply]
Thanks a lot, I will try !

**Reply by hariprasath-yadav** [reply]
> I can't downgrade without downgrade all dependencies :'(

Just try once with, only downgrading the lucid package if you are not satisfied with that then upgrade the package to the original version

**Comment by LeadcodeDev** [comment, marked as answer]
Hello! 👋 I come back to you, I tried the method of @hariprasath-yadav but downgrade lucid generates an incompatibility in the versions of Adonis (so I came back to @lasted).

I also tried to change the datetime type to timestamp but the same error persists.

Do you have another idea ?

Thanks in advance

**Reply by LeadcodeDev** [reply]
EDIT: After logging my DateTime object, I found a ts property and I'm wondering if it wouldn't be interpreted as a column ?
What do you think about it ?
![image](https://user-images.githubusercontent.com/8946317/99233748-f3750200-27f3-11eb-8026-65e2cbe2acba.png)

**Reply by thetutlage** [reply]
Can you share the exact database call you are making using the `Promotion` model?

**Reply by LeadcodeDev** [reply]
Thank you very much for your answer @thetutlage !
I just found, in my template, my datetime column had the type DateTime but this one was not declared as such....

```ts
@column()
public start: DateTime

@column()
public expiration: DateTime
```
To
```ts
@column.dateTime()
public start: DateTime

@column.dateTime()
public expiration: DateTime
```
I'm just too dumb I guess...

Thank you all for your help ! ❤
